src/protocol/models.rs

```diff
@@ -41,14 +42,57 @@ use crate::models::Token;
 /// * `tokens`: `Vec<ERC20Token>`, the tokens of the trading pair
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ProtocolComponent {
+    #[deprecated(since = "0.73.0", note = "Use `id` instead")]
     pub address: Bytes,
+    pub id: Bytes,
     pub tokens: Vec<Token>,
+    pub protocol_system: String,
+    pub protocol_type_name: String,
+    pub chain: Chain,
+    pub contract_ids: Vec<Bytes>,
+    pub static_attributes: HashMap<String, Bytes>,
+    pub creation_tx: Bytes,
+    pub created_at: NaiveDateTime,
 }
 
 impl ProtocolComponent {
-    pub fn new(address: Bytes, mut tokens: Vec<Token>) -> Self {
+    #[allow(deprecated)]
+    pub fn new(mut tokens: Vec<Token>, core_model: tycho_core::dto::ProtocolComponent) -> Self {
         tokens.sort_unstable_by_key(|t| t.address.clone());
-        ProtocolComponent { address, tokens }
+        let id = Bytes::from(core_model.id.as_str());
+        ProtocolComponent {
+            id: id.clone(),
+            address: id,
+            tokens,
+            protocol_system: core_model.protocol_system,
+            protocol_type_name: core_model.protocol_type_name,
+            chain: core_model.chain,
+            contract_ids: core_model.contract_ids,
+            static_attributes: core_model.static_attributes,
+            creation_tx: core_model.creation_tx,
+            created_at: core_model.created_at,
```

Comment:
Do we need this? Or are we adding it just in case

Reply:
adding just in case.. it's not a lot of info, so I think it's fine to have it!